lib/tasks/book2.rake

```diff
@@ -33,31 +33,40 @@ def genbook (code, &get_content)
 
   nav = '<div id="nav"><a href="[[nav-prev]]">prev</a> | <a href="[[nav-next]]">next</a></div>'
 
-  atlas = JSON.parse(get_content.call("atlas.json"))
+  progit = get_content.call("progit.asc")
 
   chapters = {}
   appnumber = 0
   chnumber = 0
   secnumber = 0
   ids = {}
 
-  atlas['files'].each_with_index do |filename, index|
-    if filename =~ /book\/[0-9].*\/1-[^\/]*\.asc/
+  chaps = progit.scan( /(book\/[0-9A-C].*\/1-[^\/]*\.asc|[01A-C].*\.asc)/).flatten
```

Comment:
can we replace the (this and a few others below) regex usage by a variable with a meaningful name (possibly with a comment explaining something more hacky)? It would help if someone help wants to fix a bug/do some improvement on this some day

Reply:
You're right, it's necessary. Let's document a little.